### PR TITLE
Feat/read-only and to-skip storage services

### PR DESCRIPTION
**Pull request by theus77**
|Q              |A  |
|---------------|---|
|Bug fix?       |N|
|New feature?   |Y|	
|BC breaks?     |Y|	
|Deprecations?  |N|	
|Fixed tickets  |N|

### CODE REVIEW

File: src/Storage/Factory/EntityFactory.php [hunk 1]
```diff
@@ -44,7 +44,7 @@ public function createService(array $parameters): ?StorageInterface
         }
         $this->registered = true;
 
-         return new EntityStorage($this->doctrine);
+         return new EntityStorage($this->doctrine, $config[self::STORAGE_CONFIG_READ_ONLY], $config[self::STORAGE_CONFIG_TO_SKIP]);
```
Comment (dameert) [inline src/Storage/Factory/EntityFactory.php:47]: Small detail, I prefer "skip" in stead of "to-skip"
you use "read-only" and not "to-be-ready-only"

File: src/Storage/Service/AbstractUrlStorage.php [hunk 2]
```diff
@@ -132,19 +161,32 @@ public function addChunk(string $hash, string $chunk): bool
...
         }
         return false;
     }
+
+    public function isReadOnly(): bool
+    {
+        return $this->readOnly;
+    }
+
+    public function isToSkip(): bool
```
Comment (dameert) [inline src/Storage/Service/AbstractUrlStorage.php:188]: This would become "shouldSkip()"

Comment (theus77) [reply src/Storage/Service/AbstractUrlStorage.php]: Agree (and it's the reason of toSkip, isSkip was weird) naming issue 96993866c927751ec0fac5a9886e27fdc4d771b8

File: src/Storage/Service/AbstractUrlStorage.php [hunk 3]
```diff
@@ -185,8 +185,8 @@ public function isReadOnly(): bool
         return $this->readOnly;
     }
 
-    public function isToSkip(): bool
+    public function isSkip(): bool
```
Comment (dameert) [inline src/Storage/Service/AbstractUrlStorage.php]:
```suggestion
    public function shouldSkip(): bool
```

"is" for property
"should" for action

File: src/Storage/Service/AbstractUrlStorage.php [hunk 4]
```diff
@@ -185,8 +185,8 @@ public function isReadOnly(): bool
...
     {
-        return $this->toSkip;
+        return $this->skip;
```
Comment (dameert) [inline src/Storage/Service/AbstractUrlStorage.php:190]:
```suggestion
        return $this->shouldSkip;
```

I wonder if its not better to also name the variable with "is" or "should" prefix.

private $readOnly; indicates less then private $isReadOnly; that we are handling a bool.
Idem for private $skip vs private $shouldSkip;

**Review by dameert — CHANGES_REQUESTED** [review]

**Pull request merged**